### PR TITLE
 [WFLY-12038]: Expose the enable1Prefixes configuration on external (pooled) connection factories.

### PR DESCRIPTION
**Pull request by ehsavoie**
* Adding a new attribute 'enable-amq1-prefix' on external connection
factories to override the use of those compatibility prefixes when connecting to a remote artemis broker.

Jira: https://issues.jboss.org/browse/WFLY-12038

Depends on PR: #12150

### CODE REVIEW

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingTransformerRegistration.java [hunk 1]
```diff
@@ -70,21 +70,27 @@ public String getSubsystemName() {
...
     }
 
-    private static void registerTransformers_WF_16(ResourceTransformationDescriptionBuilder subsystem) {
+    private static void registerTransformers_WF_17(ResourceTransformationDescriptionBuilder subsystem) {
         ResourceTransformationDescriptionBuilder server = subsystem.addChildResource(SERVER_PATH);
         rejectDefinedAttributeWithDefaultValue(server, ServerDefinition.JOURNAL_FILE_OPEN_TIMEOUT);
 
-        ResourceTransformationDescriptionBuilder queue = server.addChildResource(QUEUE_PATH);
+        rejectDefinedAttributeWithDefaultValue(subsystem.addChildResource(PathElement.pathElement(CONNECTION_FACTORY)), ConnectionFactoryAttributes.External.ENABLE_AMQ1_PREFIX);
+        rejectDefinedAttributeWithDefaultValue(subsystem.addChildResource(PathElement.pathElement(POOLED_CONNECTION_FACTORY)), ConnectionFactoryAttributes.External.ENABLE_AMQ1_PREFIX);
```
Comment (kabir) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingTransformerRegistration.java]: I have checked these with Emmanuel and these are correct

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_Connect_a_pooled-connection-factory_to_a_Remote_Artemis_Server.adoc [hunk 2]
```diff
@@ -32,6 +32,13 @@ created at step (2).
 /subsystem=messaging-activemq/pooled-connection-factory=remote-artemis:add(connectors=[remote-artemis], entries=[java:/jms/remoteCF])
 ----
 
+Please note that if you are connecting to an Artemis 2.x the old prefixes may not be used anymore for queues and topics.
```
Comment (bstansberry) [inline docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_Connect_a_pooled-connection-factory_to_a_Remote_Artemis_Server.adoc]: 1) This should explain what's meant by these prefixes, an example or something. It's not covered earlier in the document.
2) This should state that by default WildFy will prepend the prefix.

**Review by jamezp — APPROVED** [review]

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_Connect_a_pooled-connection-factory_to_a_Remote_Artemis_Server.adoc [hunk 3]
```diff
@@ -32,6 +32,15 @@ created at step (2).
 /subsystem=messaging-activemq/pooled-connection-factory=remote-artemis:add(connectors=[remote-artemis], entries=[java:/jms/remoteCF])
 ----
 
+In Artemis 1.x topics and queues used had a prefix(jms.topic. and jms.queue.) that were prepended to the destiantion name.
```
Comment (bstansberry) [inline docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_Connect_a_pooled-connection-factory_to_a_Remote_Artemis_Server.adoc]: s/used had/used to have/g
s/destiantion/destination

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_Connect_a_pooled-connection-factory_to_a_Remote_Artemis_Server.adoc [hunk 4]
```diff
@@ -32,6 +32,15 @@ created at step (2).
...
+In Artemis 2.x this is no longer the case, but for compatibility reasons WildFly still prepend those prefixes and tells Artemis to run in compatibility mode.
```
Comment (bstansberry) [inline docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_Connect_a_pooled-connection-factory_to_a_Remote_Artemis_Server.adoc]: s/prepend/prepends

should this be "tells any embedded Artemis server to run in compatibility mode"?  So WF deals with embedded but with remote Artemis you may need this setting.

**Review by bstansberry — CHANGES_REQUESTED** [review]

**Review by bstansberry — APPROVED** [review]
I still have an unaddressed question about the doc but that can come later.

**Pull request merged**